perception/lidar_transfusion/src/lidar_transfusion_node.cpp

```diff
@@ -0,0 +1,161 @@
+      this->declare_parameter<std::vector<std::string>>("iou_nms_target_class_names");
+    p.search_distance_2d_ = this->declare_parameter<double>("iou_nms_search_distance_2d");
+    p.iou_threshold_ = this->declare_parameter<double>("iou_nms_threshold");
+    iou_bev_nms_.setParameters(p);
+  }
+  const auto yaw_norm_thresholds =
+    this->declare_parameter<std::vector<double>>("yaw_norm_thresholds");
+  const float score_threshold =
+    static_cast<float>(this->declare_parameter<double>("score_threshold", 0.2));
+
+  NetworkParam network_param(onnx_path, engine_path, trt_precision);
+  DensificationParam densification_param(
+    densification_world_frame_id, densification_num_past_frames);
+  TransfusionConfig config(
+    voxels_num, point_cloud_range, voxel_size, circle_nms_dist_threshold, yaw_norm_thresholds,
+    score_threshold);
+
+  const auto allow_remapping_by_area_matrix =
+    this->declare_parameter<std::vector<int64_t>>("allow_remapping_by_area_matrix");
+  const auto min_area_matrix = this->declare_parameter<std::vector<double>>("min_area_matrix");
+  const auto max_area_matrix = this->declare_parameter<std::vector<double>>("max_area_matrix");
+  detection_class_remapper_.setParameters(
+    allow_remapping_by_area_matrix, min_area_matrix, max_area_matrix);
+
+  detector_ptr_ = std::make_unique<TransfusionTRT>(network_param, densification_param, config);
+
+  cloud_sub_ = this->create_subscription<sensor_msgs::msg::PointCloud2>(
+    "~/input/pointcloud", rclcpp::SensorDataQoS{}.keep_last(1),
+    std::bind(&LidarTransfusionNode::cloudCallback, this, std::placeholders::_1));
+
+  objects_pub_ = this->create_publisher<autoware_auto_perception_msgs::msg::DetectedObjects>(
+    "~/output/objects", rclcpp::QoS(1));
+
+  published_time_pub_ = std::make_unique<tier4_autoware_utils::PublishedTimePublisher>(this);
+
+  // initialize debug tool
+  {
+    using tier4_autoware_utils::DebugPublisher;
+    using tier4_autoware_utils::StopWatch;
+    stop_watch_ptr_ = std::make_unique<StopWatch<std::chrono::milliseconds>>();
+    debug_publisher_ptr_ = std::make_unique<DebugPublisher>(this, this->get_name());
+    stop_watch_ptr_->tic("cyclic");
+    stop_watch_ptr_->tic("processing/total");
+  }
+
+  if (this->declare_parameter("build_only", false)) {
+    RCLCPP_INFO(this->get_logger(), "TensorRT engine is built and shutdown node.");
+    rclcpp::shutdown();
+  }
+}
+
+void LidarTransfusionNode::cloudCallback(const sensor_msgs::msg::PointCloud2::ConstSharedPtr msg)
+{
+  const auto objects_sub_count =
+    objects_pub_->get_subscription_count() + objects_pub_->get_intra_process_subscription_count();
+  if (objects_sub_count < 1) {
+    return;
+  }
+
+  if (stop_watch_ptr_) {
+    stop_watch_ptr_->toc("processing/total", true);
+  }
+
+  std::vector<Box3D> det_boxes3d;
+  std::unordered_map<std::string, double> proc_timing;
+  bool is_success = detector_ptr_->detect(*msg, tf_buffer_, det_boxes3d, proc_timing);
```

Comment:
This was already a code smell in centerpoint, but we are adding a new parameter, making it more noticeable.
Instead of using references as arguments to return values, it is recommended to use tuples.
Then, in this file, we can use structured bindings like: `auto & [is_success, det_boxes3d, proc_timing]= ...`

Reply:
As we discussed, RVO / NRVO will not apply in that case and avoiding the copy with structured bindings seems not trivial here. Since the current approach is working, we will leave possible improvement for future PR.